documentation/commands.md

```diff
@@ -1,233 +1,243 @@
-          {
-            "type": "dataset",
-            "name": "string_scalar",
-            "dataset": {
-              "type": "string"
-            },
-            "values": "the-scalar-string"
-          },
-          {
-            "type": "dataset",
-            "name": "string_3d",
-            "dataset": {
-              "type": "string",
-              "size": ["unlimited", 3, 2]
-            },
-            "values": [
-              [
-                ["string_0_0_0", "string_0_0_1"],
-                ["string_0_1_0", "string_0_1_1"],
-                ["string_0_2_0", "string_0_2_1"]
-              ],
-              [
-                ["string_1_0_0", "string_1_0_1"],
-                ["string_1_1_0", "string_1_1_1"],
-                ["string_1_2_0", "string_1_2_1"]
-              ]
-            ]
-          },
+          }
+        ],
+        "attributes": [
           {
-            "type": "dataset",
-            "name": "string_fixed_length_1d",
-            "dataset": {
-              "type":"string",
-              "string_size": 32,
-              "size": ["unlimited"]
-            },
-            "values": ["the-scalar-string", "another-one"],
-            "attributes": [
-            {
-              "name": "scalar_attribute",
-              "values": 42
-            },
-            {
-              "name": "vector_attribute",
-              "values": [1, 2, 3],
-              "type": "uint32"
-            }
-            ]
+            "name": "units",
+            "values": "ms"
           }
         ]
       }
     ]
-  },
-  "file_attributes": {
-    "file_name": "some.h5"
-  },
-  "cmd": "FileWriter_new",
-  "job_id": "unique-identifier",
-  "broker": "localhost:9092",
-  "start_time": "[OPTIONAL] timestamp (int) in milliseconds",
-  "stop_time": "[OPTIONAL] timestamp (int) in milliseconds",
-  "service_id": "[OPTIONAL] the_name_of_the_instance_which_should_interpret_this_command",
-  "abort_on_uninitialised_stream": false,
-  "use_hdf_swmr": true
 }
 ```
-By default, the file-writer will continue to write a file even when a stream fails to initialise.
-In order to abort writing instead `set abort_on_uninitialised_stream` to `true`.
 
-#### Command to exit the file-writer
 
-```json
-{
-  "cmd": "FileWriter_exit",
-  "service_id": "[OPTIONAL] the_name_of_the_instance_which_should_interpret_this_command"
-}
-```
+In the example above, a group called my_test_group is defined which in turn has a stream and an attribute defined.
+The stream definition instructs the file-writer that there is data available which should be written to the containing group.
+The parameters for the stream definition are:
 
-#### Command to stop a file being written
+- writer_module: The FlatBuffers schema in the [streaming-data-types](https://github.com/ess-dmsc/streaming-data-types) repository that was used to serialise the data.
+- type: The type of the data. The possible types are defined in the schema declared in the `writer_module`.
```

Comment:
We should have some information on the different requirements of `type` for the different writer modules but maybe that is outside the scope of this PR.

Reply:
Agreed on both counts